tests/test_bookmark.py

```diff
@@ -0,0 +1,163 @@
+        Verify that for full table stream, all data replicated in sync 1 is replicated again in sync 2.
+        PREREQUISITE
+        For EACH stream that is incrementally replicated there are multiple rows of data with
+            different values for the replication key
+        """
+        
+        
+        expected_streams = self.expected_streams()
+        expected_replication_keys = self.expected_replication_keys()
+        expected_replication_methods = self.expected_replication_method()
+
+        ##########################################################################
+        # First Sync
+        ##########################################################################
+        conn_id = connections.ensure_connection(self)
+
+        # Run in check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # table and field selection
+        catalog_entries = [catalog for catalog in found_catalogs
+                           if catalog.get('tap_stream_id') in expected_streams]
+
+        self.perform_and_verify_table_and_field_selection(
+            conn_id, catalog_entries)
+
+        # Run a first sync job using orchestrator
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        # Update State Between Syncs
+        ##########################################################################
+        
+        LOGGER.info("Current Bookmark: {}".format(first_sync_bookmarks))
+        new_states = menagerie.get_state(conn_id)
+        new_states = {'currently_syncing': None,
+                      'bookmarks': {'conversations': {list(expected_replication_keys['conversations'])[0]: 1655856000000},
+                                    'messages': {list(expected_replication_keys['messages'])[0]: 1655856000000},
+                                    'activity_logs': {list(expected_replication_keys['activity_logs'])[0]: "2022-06-22T00:00:00Z"}}}
+
+        menagerie.set_state(conn_id, new_states)
+        LOGGER.info("New Bookmark: {}".format(menagerie.get_state(conn_id)))
+
+        ##########################################################################
+        # Second Sync
+        ##########################################################################
+
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        # Third Sync: 
+        #   To test that we get at least one record per replication
+        #   Assuming that there are new records between 2nd and 3rd sync
```

Comment:
Based on the final assertion I don't think this is assuming anything about records between syncs 2 and 3. If no records are inserted or updated than we get 1 record. If they are we get more than 1. I think if this comment should be true, then the assertion on line 161 should be removed.